python/strings_udf/cpp/include/cudf/strings/udf/strip.cuh

```diff
@@ -0,0 +1,110 @@
+
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "udf_string.cuh"
+
+#include <cudf/strings/detail/strip.cuh>
+#include <cudf/strings/string_view.cuh>
+
+namespace cudf {
+namespace strings {
+namespace udf {
+
+/**
+ * @brief Strip characters from the beginning and/or end of the given string
+ *
+ * The `d_to_strip` is interpreted as an array of characters to be removed.
+ * If `d_to_strip` is an empty string, whitespace characters are stripped.
+ *
+ * @code{.cpp}
+ * auto d_str = cudf::string_view{" aba ", 5};
+ * auto d_to_strip = cudf::string_view{}; // empty string
+ * auto result = strip(d_str, d_to_strip);
+ * // result is "aba"
+ * d_to_strip = cudf::string_view{" a", 2}; // space and 'a'
+ * result = strip(d_str, d_to_strip);
+ * // result is "b" ('a' or ' ' removed from the ends)
+ * @endcode
+ *
+ * @param d_str String to strip characters from
+ * @param d_to_strip Characters to remove
+ * @param stype From where to strip the characters;
+ *              Default `BOTH` indicates stripping characters from the
+ *              beginning and the end of the input string `d_str`
+ * @return New string with characters removed
+ */
+__device__ udf_string strip(cudf::string_view const d_str,
```

Comment:
I don't agree that would be better than this.

Reply:
Is there any standard for what `strip` should look like? I see that it's not a `std::string` method, so if we are just implementing our own that is fine, but for context in Python we have `strip`, `lstrip`, and `rstrip`, where `strip` does not take any extra arguments to indicate a side. The argument is to avoid having multiple ways to do something without there being a clear reason to prefer one or the other, but that is obviously more of a Python mindset than a C++ one in this instance.

Reply:
The default parameter is there so you don't have to specify a side-type. This takes advantage of that on purpose. I guess you are saying why would someone call `strip` without wanting `BOTH`. I think that makes sense. The intention was to provide only a `strip` method but I think I was asked to add the `lstrip` and `rstrip` later for parity with cudf/udf.
Would it be acceptable to remove the `lstrip` and `rstrip` functions?

Reply:
Yes I would be fine with that and just having the Python call `strip` with the appropriate `side_type`.